packages/react/src/components/TreeView/TreeView-test.js

```diff
@@ -1,95 +1,219 @@
 /**
- * Copyright IBM Corp. 2020
+ * Copyright IBM Corp. 2022
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
 
-import { CaretDown } from '@carbon/icons-react';
 import React from 'react';
-import { mount } from 'enzyme';
-import { Document, Folder } from '@carbon/icons-react';
-import TreeView, { TreeNode } from './';
-
-const prefix = 'cds';
+import TreeView from './TreeView';
+import TreeNode from './TreeNode';
+import userEvent from '@testing-library/user-event';
+import { fireEvent, render, screen, within } from '@testing-library/react';
+import { Document } from '@carbon/icons-react';
 
 describe('TreeView', () => {
-  let wrapper;
-  let onTreeSelect;
-  let onNodeSelect;
-
-  beforeEach(() => {
-    onTreeSelect = jest.fn();
-    onNodeSelect = jest.fn();
-    wrapper = mount(
-      <TreeView label="Tree view" selected={['1']} onSelect={onTreeSelect}>
-        <TreeNode id="1" value="1" label="1" />
-        <TreeNode id="2" value="2" label="2" onSelect={onNodeSelect} />
-        <TreeNode id="5" value="5" label="5" isExpanded>
-          <TreeNode id="5-1" value="5-1" label="5-1" />
-          <TreeNode id="5-2" value="5-2" label="5-2" />
-          <TreeNode id="5-3" value="5-3" label="5-3" isExpanded>
-            <TreeNode id="5-4" value="5-4" label="5-4" />
+  describe('renders as expected - Component API', () => {
+    it('should spread extra props onto outermost element', () => {
+      render(<TreeView data-testid="test-id" label="Tree" />);
+
+      expect(screen.getByRole('tree')).toHaveAttribute(
+        'data-testid',
+        'test-id'
+      );
+    });
+
+    it('should respect active prop', () => {
+      render(
+        <TreeView active="Node 1" label="Tree">
+          <TreeNode data-testid="Node 1" id="Node 1" label="Node 1" />
+          <TreeNode id="Node 2" label="Node 2" />
+        </TreeView>
+      );
+
+      const node = screen.getByTestId('Node 1');
+
+      expect(node).toHaveClass('cds--tree-node--active');
+    });
+
+    it('should render children as expected', () => {
+      render(
+        <TreeView label="Tree View">
+          <TreeNode isExpanded={true} data-testid="Node 1" label="Node 1">
+            <TreeNode data-testid="Node 2" label="Node 2" />
           </TreeNode>
-        </TreeNode>
-      </TreeView>
-    );
-  });
+        </TreeView>
+      );
 
-  it('should render', () => {
-    expect(wrapper).toMatchSnapshot();
-  });
+      const nodeParent = screen.getByTestId('Node 1');
+      const nodeChild = screen.getByTestId('Node 2');
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(nodeParent).toHaveClass('cds--tree-parent-node');
+      expect(nodeChild).toHaveClass('cds--tree-leaf-node');
+      expect(within(nodeParent).getByText('Node 1')).toBeInTheDocument();
+      expect(within(nodeChild).getByText('Node 2')).toBeInTheDocument();
```

Comment:
What's the difference between using `within()` like this vs the native `nodeParent.contains(nodeChild)`?

Reply:
From what i understand, `within` will return a....container?, where it can be returned from other rtl queries and search for all the queries within in like this example 

```
 takes a DOM element and binds it to the raw query functions, allowing them to be used without specifying a container
```

https://testing-library.com/docs/dom-testing-library/api-within/

where the old way seems to be more of if this container is inside a parent one without considering anything else

could this both work? maybe, i just went off the RTl way of doing it tbh